From 778b8f21ba0b71289aef37e3422d008456445971 Mon Sep 17 00:00:00 2001
From: Frank Heckenbach
Date: Fri, 24 May 2019 23:58:47 +0200
Subject: [PATCH 06/14] src/FTFont/FTBufferFont.cpp,
 src/FTFont/FTTextureFont.cpp: GL_TEXTURE_ENV_MODE is not valid mask for
 glPushAttrib. Use GL_TEXTURE_BIT instead to avoid leaking texture env mode.
 (reported by Eddie-cz, https://github.com/frankheckenbach/ftgl/issues/3)

---
 src/FTFont/FTBufferFont.cpp  | 2 +-
 src/FTFont/FTTextureFont.cpp | 2 +-
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/FTFont/FTBufferFont.cpp b/src/FTFont/FTBufferFont.cpp
index ce04cf56ae0f..b330a3b119f7 100644
--- a/src/FTFont/FTBufferFont.cpp
+++ b/src/FTFont/FTBufferFont.cpp
@@ -230,11 +230,11 @@ inline FTPoint FTBufferFontImpl::RenderI(const T* string, const int len,
     int width, height, texWidth, texHeight;
     int cacheIndex = -1;
     bool inCache = false;

     // Protect blending functions, GL_TEXTURE_2D and optionally GL_BLEND
-    glPushAttrib(GL_COLOR_BUFFER_BIT | GL_ENABLE_BIT | GL_TEXTURE_ENV_MODE);
+    glPushAttrib(GL_COLOR_BUFFER_BIT | GL_ENABLE_BIT | GL_TEXTURE_BIT);

     // Protect glPixelStorei() calls
     glPushClientAttrib(GL_CLIENT_PIXEL_STORE_BIT);

     if(FTLibrary::Instance().GetLegacyOpenGLStateSet())
diff --git a/src/FTFont/FTTextureFont.cpp b/src/FTFont/FTTextureFont.cpp
index 97e876876e12..a5145cf05f53 100644
--- a/src/FTFont/FTTextureFont.cpp
+++ b/src/FTFont/FTTextureFont.cpp
@@ -239,11 +239,11 @@ template <typename T>
 inline FTPoint FTTextureFontImpl::RenderI(const T* string, const int len,
                                           FTPoint position, FTPoint spacing,
                                           int renderMode)
 {
     // Protect GL_TEXTURE_2D and optionally GL_BLEND
-    glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT | GL_TEXTURE_ENV_MODE);
+    glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT | GL_TEXTURE_BIT);

     if(FTLibrary::Instance().GetLegacyOpenGLStateSet())
       {
         glEnable(GL_BLEND);
         /*
--
2.47.1


